fuel/app/classes/materia/api/v1.php

```diff
@@ -68,6 +68,46 @@ static public function widget_instance_delete($inst_id)
 		return $inst->db_remove();
 	}
 
+	/**
+	 * @return object, contains properties indicating whether the current
+	 * user can edit the widget and a message object describing why, if not
+	 */
+	static public function widget_instance_edit_perms_verify(string $inst_id): \stdClass
+	{
+		$response = new \stdClass();
```

Comment:
Style wise - I'd move this initialization below the validations that follow

Reply:
This method will always return a `stdClass` and the validations all potentially modify `$response->msg`, so it'd have to be defined beforehand. The alternative is to have a `$msg` that we define at the top and have all the validations set that instead, then define `$response` afterwards, but that seems like it'd be unnecessary.